### PR TITLE
Update Mosviz to use Spectrum1D from SpectralCube

### LINKED ISSUE #837: TST: SpectralGWCS failures using specutils 1.3.1 and glue-astronomy dev
Example log: https://github.com/spacetelescope/jdaviz/runs/3508608981?check_suite_focus=true

```
Package versions: 
Numpy: 1.22.0.dev0+949.ga90677a0b
Scipy: 1.7.1
Matplotlib: 3.4.3
h5py: 3.4.0
Pandas: 1.3.2
astropy: 5.0.dev786+g0fd7ae818
pyyaml: 5.4.1
scikit-image: 0.18.3
specutils: 1.3.1
spectral-cube: 0.5.1.dev227+g8d07dc9
asteval: 0.9.25
click: 8.0.1
echo: 0.5
idna: 2.10
traitlets: 5.1.0
bqplot: 0.12.30
bqplot-image-gl: 1.4.4
glue-core: 1.2.1
glue-jupyter: 0.8
glue-astronomy: 0.3.dev15+g312e959
ipyvue: 1.5.0
ipyvuetify: 1.8.1
ipysplitpanes: 0.2.0
ipygoldenlayout: 0.4.0
voila: 0.2.11
vispy: 0.8.1
gwcs: 0.16.2a1.dev32+g464e55f
asdf: 2.9.0.dev11+g9e64fbe
```

```
____________________________ test_viewer_axis_link _____________________________

mosviz_app = <jdaviz.configs.mosviz.helper.MosViz object at 0x7f623bc65e20>
spectrum1d = <Spectrum1D(flux=<Quantity [ 9.29172181, 11.92673337, 13.19606653, ..., 16.03141721,
           15.67835212, 17.780023...tralAxis [6000.        , 6001.95503421, 6003.91006843, ..., 7996.08993157,
   7998.04496579, 8000.        ] Angstrom>)>
spectrum2d = SpectralCube with shape=(15, 1, 1024):
 n_x:   1024  type_x: RA---TAN  unit_x: deg    range:     0.999898 deg:   86.79...:     0.055866 deg:    0.999746 deg
 n_s:     15  type_s: WAVE      unit_s: m      range:        1.000 m:      15.000 m

    def test_viewer_axis_link(mosviz_app, spectrum1d, spectrum2d):
        label1d = "Test 1D Spectrum"
        mosviz_app.load_1d_spectra(spectrum1d, data_labels=label1d)
    
        label2d = "Test 2D Spectrum"
        mosviz_app.load_2d_spectra(spectrum2d, data_labels=label2d)
    
        table = mosviz_app.app.get_viewer('table-viewer')
        table.widget_table.vue_on_row_clicked(0)
    
        scale_2d = mosviz_app.app.get_viewer('spectrum-2d-viewer').scales['x']
        scale_1d = mosviz_app.app.get_viewer('spectrum-viewer').scales['x']
    
        scale_2d.min = 200.0
>       assert scale_1d.min == spectrum1d.spectral_axis.value[200]
E       assert 6000.0 == 6391.006842619746
E        +  where 6000.0 = LinearScale(allow_padding=False, max=6000.0, min=6000.0).min

../../.tox/py39-test-devdeps/lib/python3.9/site-packages/jdaviz/configs/mosviz/tests/test_helper.py:261: AssertionError
________________________ test_get_spectral_regions_unit ________________________

specviz_app = <jdaviz.configs.specviz.helper.SpecViz object at 0x7f623a348eb0>
spectrum1d = <Spectrum1D(flux=<Quantity [12.49671415, 12.30618014, 13.53658529, 14.85657834, 13.5436244 ,
           13.98808527, 1...28024,
                   1.72491783, 0.56228753, 1.01283112, 0.31424733,
                   0.90802408, 1.4123037 ]))>

    def test_get_spectral_regions_unit(specviz_app, spectrum1d):
        # Ensure units we put in are the same as the units we get out
        specviz_app.load_spectrum(spectrum1d)
        specviz_app.app.get_viewer("spectrum-viewer").apply_roi(XRangeROI(1, 3.5))
    
>       subsets = specviz_app.get_spectral_regions()

../../.tox/py39-test-devdeps/lib/python3.9/site-packages/jdaviz/configs/specviz/tests/test_helper.py:88: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../.tox/py39-test-devdeps/lib/python3.9/site-packages/jdaviz/configs/specviz/helper.py:78: in get_spectral_regions
    regions = self.app.get_subsets_from_viewer("spectrum-viewer", subset_type="spectral")
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = Application(components={'g-viewer-tab': '<template>\n  <component :is="stack.container">\n    <g-viewer-tab\n      v-f...lbackList with 1 elements>
  tool_items: <CallbackList with 3 elements>
  tray_items: <CallbackList with 5 elements>
>)
viewer_reference = 'spectrum-viewer', data_label = None
subset_type = 'spectral'

    def get_subsets_from_viewer(self, viewer_reference, data_label=None, subset_type=None):
        """
        Returns the subsets of a specified viewer converted to astropy regions
        objects.
    
        It should be noted that the subset translation machinery lives in the
        glue-astronomy repository. Currently, the machinery only works on 2D
        data for cases like range selection. For e.g. a profile viewer that is
        ostensibly just a view into a 3D data set, it is necessary to first
        reduce the dimensions of the data, then retrieve the subset information
        as a regions object. This means that the returned y extents in the
        region are not strictly representative of the subset range in y.
    
        Parameters
        ----------
        viewer_reference : str
            The reference to the viewer defined with the ``reference`` key
            in the yaml configuration file.
        data_label : str, optional
            Optionally provide a label to retrieve a specific data set from the
            viewer instance.
        subset_type : str, optional
            Optionally specify either "spectral" or "spatial" to return only
            subsets created in a profile (spectrum) viewer or image viewer,
            respectively.
    
        Returns
        -------
        data : dict
            A dict of the transformed Glue subset objects, with keys
            representing the subset name and values as astropy regions
            objects.
        """
        viewer = self.get_viewer(viewer_reference)
        data = self.get_data_from_viewer(viewer_reference,
                                         data_label,
                                         cls=None)
        regions = {}
    
        if data_label is not None:
            data = {data_label: data}
    
        for key, value in data.items():
            if isinstance(value, Subset):
                # Skip spatial or spectral subsets if only the other is wanted
                if subset_type == "spectral" and isinstance(value.subset_state, RoiSubsetState):
                    continue
                elif subset_type == "spatial" and isinstance(value.subset_state, RangeSubsetState):
                    continue
    
                # Range selection on a profile is currently not supported in
                #  the glue translation machinery for astropy regions, so we
                #  have to do it manually. Only data that is 2d is supported,
                #  therefore, if the data is already 2d, simply use as is.
                if value.data.ndim == 2:
                    region = value.data.get_selection_definition(
                        subset_id=key, format='astropy-regions')
                    regions[key] = region
                    continue
                # There is a special case for 1d data (which is also not
                #  supported currently). We now eschew the use of the
                #  translation machinery entirely and construct the astropy
                #  region ourselves.
                elif value.data.ndim == 1:
                    # Grab the data units from the glue-astronomy spectral axis
                    # TODO: this needs to be much simpler; i.e. data units in
                    #  the glue component objects
>                   unit = value.data.coords.spectral_axis.unit
E                   AttributeError: 'SpectralGWCS' object has no attribute 'spectral_axis'

../../.tox/py39-test-devdeps/lib/python3.9/site-packages/jdaviz/app.py:523: AttributeError
__________________ test_get_spectral_regions_unit_conversion ___________________

specviz_app = <jdaviz.configs.specviz.helper.SpecViz object at 0x7f623a28c250>
spectrum1d = <Spectrum1D(flux=<Quantity [12.49671415, 12.30618014, 13.53658529, 14.85657834, 13.5436244 ,
           13.98808527, 1...28024,
                   1.72491783, 0.56228753, 1.01283112, 0.31424733,
                   0.90802408, 1.4123037 ]))>

    def test_get_spectral_regions_unit_conversion(specviz_app, spectrum1d):
        # If the reference (visible) data changes via unit conversion,
        # check that the region's units convert too
        specviz_app.load_spectrum(spectrum1d)
        specviz_app.app.get_viewer("spectrum-viewer").apply_roi(XRangeROI(1, 3.5))
    
        # Convert the wavelength axis to microns
        new_spectral_axis = "micron"
        conv_func = uc.UnitConversion.process_unit_conversion
        converted_spectrum = conv_func(specviz_app.app, spectrum=spectrum1d,
                                       new_spectral_axis=new_spectral_axis)
    
        # Add this new data and clear the other, making the converted spectrum our reference
        specviz_app.app.add_data(converted_spectrum, "Converted Spectrum")
        specviz_app.app.add_data_to_viewer("spectrum-viewer",
                                           "Converted Spectrum",
                                           clear_other_data=True)
    
        # Retrieve the Subset
>       subsets = specviz_app.get_spectral_regions()

../../.tox/py39-test-devdeps/lib/python3.9/site-packages/jdaviz/configs/specviz/tests/test_helper.py:114: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../.tox/py39-test-devdeps/lib/python3.9/site-packages/jdaviz/configs/specviz/helper.py:78: in get_spectral_regions
    regions = self.app.get_subsets_from_viewer("spectrum-viewer", subset_type="spectral")
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = Application(components={'g-viewer-tab': '<template>\n  <component :is="stack.container">\n    <g-viewer-tab\n      v-f...lbackList with 1 elements>
  tool_items: <CallbackList with 3 elements>
  tray_items: <CallbackList with 5 elements>
>)
viewer_reference = 'spectrum-viewer', data_label = None
subset_type = 'spectral'

    def get_subsets_from_viewer(self, viewer_reference, data_label=None, subset_type=None):
        """
        Returns the subsets of a specified viewer converted to astropy regions
        objects.
    
        It should be noted that the subset translation machinery lives in the
        glue-astronomy repository. Currently, the machinery only works on 2D
        data for cases like range selection. For e.g. a profile viewer that is
        ostensibly just a view into a 3D data set, it is necessary to first
        reduce the dimensions of the data, then retrieve the subset information
        as a regions object. This means that the returned y extents in the
        region are not strictly representative of the subset range in y.
    
        Parameters
        ----------
        viewer_reference : str
            The reference to the viewer defined with the ``reference`` key
            in the yaml configuration file.
        data_label : str, optional
            Optionally provide a label to retrieve a specific data set from the
            viewer instance.
        subset_type : str, optional
            Optionally specify either "spectral" or "spatial" to return only
            subsets created in a profile (spectrum) viewer or image viewer,
            respectively.
    
        Returns
        -------
        data : dict
            A dict of the transformed Glue subset objects, with keys
            representing the subset name and values as astropy regions
            objects.
        """
        viewer = self.get_viewer(viewer_reference)
        data = self.get_data_from_viewer(viewer_reference,
                                         data_label,
                                         cls=None)
        regions = {}
    
        if data_label is not None:
            data = {data_label: data}
    
        for key, value in data.items():
            if isinstance(value, Subset):
                # Skip spatial or spectral subsets if only the other is wanted
                if subset_type == "spectral" and isinstance(value.subset_state, RoiSubsetState):
                    continue
                elif subset_type == "spatial" and isinstance(value.subset_state, RangeSubsetState):
                    continue
    
                # Range selection on a profile is currently not supported in
                #  the glue translation machinery for astropy regions, so we
                #  have to do it manually. Only data that is 2d is supported,
                #  therefore, if the data is already 2d, simply use as is.
                if value.data.ndim == 2:
                    region = value.data.get_selection_definition(
                        subset_id=key, format='astropy-regions')
                    regions[key] = region
                    continue
                # There is a special case for 1d data (which is also not
                #  supported currently). We now eschew the use of the
                #  translation machinery entirely and construct the astropy
                #  region ourselves.
                elif value.data.ndim == 1:
                    # Grab the data units from the glue-astronomy spectral axis
                    # TODO: this needs to be much simpler; i.e. data units in
                    #  the glue component objects
>                   unit = value.data.coords.spectral_axis.unit
E                   AttributeError: 'SpectralGWCS' object has no attribute 'spectral_axis'

../../.tox/py39-test-devdeps/lib/python3.9/site-packages/jdaviz/app.py:523: AttributeError
=========================== short test summary info ============================
FAILED ../../.tox/py39-test-devdeps/lib/python3.9/site-packages/jdaviz/configs/mosviz/tests/test_helper.py::test_viewer_axis_link
FAILED ../../.tox/py39-test-devdeps/lib/python3.9/site-packages/jdaviz/configs/specviz/tests/test_helper.py::test_get_spectral_regions_unit
FAILED ../../.tox/py39-test-devdeps/lib/python3.9/site-packages/jdaviz/configs/specviz/tests/test_helper.py::test_get_spectral_regions_unit_conversion
```

[🐱](https://jira.stsci.edu/browse/JDAT-1732)

### PR DESCRIPTION
**Pull request by duytnguyendtn**
This PR provides a first pass to trying to add Specutils support (Both Spectrum1D and SpectrumList) to our Mosviz parsers. It is blocked by #547. There are a few failure modes, mainly surrounding WCS indexing:

In glue core, axis name retrieval fails: 
https://github.com/glue-viz/glue/blob/ad14432229255dc31c5756434cc18bf14657c92b/glue/core/coordinate_helpers.py#L167

Likewise: 
https://github.com/glue-viz/glue/blob/ad14432229255dc31c5756434cc18bf14657c92b/glue/core/component.py#L220

I wonder if that means the WCS isn't being parsing correctly, but I'm going to need more help to diagnose this; I think this is as far as I can push solo.

Fix #837 (the part that will pin `glue-astronomy` to 0.3 or later; Tom R said he is going to put it in as part of this PR)

### CODE REVIEW

**Comment by astrofrog** [comment]
I've pushed a commit making use of the latest dev glue-astronomy - things aren't working quite right yet but at least the x and y world axes for the 2D spectrum are now correct by default (Wavelength and Offset). I'm running into issues related to the linking of the x axes of the 2D and 1D spectral viewers which I'll try and take a look at unless this is already being worked on.

**Comment by pllim** [comment]
> I'm running into issues related to the linking of the x axes of the 2D and 1D spectral viewers which I'll try and take a look at unless this is already being worked on.

I am not working on anything like that myself. @rosteen , @javerbukh , or @ojustino ?

**Comment by astrofrog** [comment]
Ok actually the axis linking issue is not crucial to reviewing this in the end. I think this should now be ready for testing out - note that this needs the latest dev version of glue-astronomy. With this PR, the axes for the 2D spectra are as follows:

Screenshot 2021-09-03 at 23 00 05

I haven't tested how the linking of datasets is working now - I think it would make sense to first wrap up https://github.com/spacetelescope/jdaviz/pull/762 then I can rebase this on top of ``main`` and make sure the linking works properly then.

I don't expect this to work flawlessly at the moment as I haven't used MOSViz enough to know what is expected to work etc - so this would really benefit from some detailed testing from MOSViz experts to determine whether anything is broken compared to ``main``.

Once we are happy with this, I can tag a release of glue-astronomy and pin the minimum version in jdaviz.

**Comment by astrofrog** [comment]
(I'll look into the test failures early next week but I don't think they are critical for testing this out)

**Comment by astrofrog** [comment]
@ojustino - with this PR, the spatial axis of the 2D spectra is no longer RA/Dec so definitely can't be linked with the images.

File: jdaviz/configs/mosviz/plugins/parsers.py [hunk 1]
```diff
@@ -532,27 +502,15 @@ def mos_niriss_parser(app, data_dir, obs_label=""):
...
-                        wcs = WCS(header)
-
-                        spec2d = SpectralCube(new_data, wcs=wcs, meta=meta)
+
+                        wcs = WCS(naxis=1)
+                        wcs.wcs.ctype = ['WAVE']
+                        wcs.wcs.crpix = [0]
+                        wcs.wcs.crval = [0]
+                        wcs.wcs.cdelt = [1.e-6]
+                        wcs.wcs.cunit = ['m']
```
Comment (astrofrog) [inline jdaviz/configs/mosviz/plugins/parsers.py:511]: Note that this is still a fake spectral WCS, as it was before.

**Comment by astrofrog** [comment]
I think the remaining CI failures may be fixed once glue-astronomy is released.

**Comment by ojustino** [comment]
@astrofrog, I was able to test Mosviz with this branch and the latest version of `glue-astronomy`. I didn't encounter anything strange with data loading, subset selection, or model fitting compared to `main`, and I do see the new axis in the Viewer menu.

**Comment by astrofrog** [comment]
Great! So maybe the best path forward is to merge #762, rebase this, release glue-astronomy then update this PR to update the minimum version of glue-astronomy - and then make sure all the subset stuff works as expected before merging this.

File: jdaviz/configs/mosviz/plugins/parsers.py (hunk 1 above)
Comment (duytnguyendtn) [reply jdaviz/configs/mosviz/plugins/parsers.py:511]: Is there a reason why we still need to fake the WCS @astrofrog? Particularly with #735's strategy to link by pixel, I imagine a WCS, even for Mosviz, is no longer an absolute requirement? I think we were hoping that with transitioning to Spectrum1D, we could drop these fake values?

My understanding was that the WCS was required to construct a SpectralCube, hence why faking these values were required. I don't believe Spectrum1Ds have this requirement?

**Comment by astrofrog** [comment]
I've now updated this to use the latest glue-astronomy version, and have started to investigate subset issues:

* When selecting subsets in a 1D spectrum, an error used to occur which meant the subset was not immediately shown but required changing object and changing back. This should be fixed by https://github.com/glue-viz/glue-jupyter/pull/256 which I'll merge shortly and release a new glue-jupyter version.

* When switching to a different 1D spectrum, the subset was not shown - it was only shown for the original spectrum the selection was made in. This is because the linking in #762 only links the spectral axis of each 1D spectrum with its corresponding 2D spectrum but didn't link each 1D spectrum to each other 1D spectrum. I've now fixed this in ab0e243058c1a672cbe89aa53b3a5a3d6690662d in this PR.

* When making a selection in a 1D spectrum, the corresponding region is not shown in the 2D spectrum. There is something funky going on with the spectral WCS of the Spectrum1D objects, which I will investigate next. Once I have fixed this, the subset propagation from 1D to 2D should work.

* When making a selection in a 2D spectrum, the corresponding region is not shown in the 1D spectrum. This is expected, as the subsets made in the 2D viewer are 2D and can't be applied to the 1D spectrum conceptually. For instance, if I only select a rectangular region that only covers a narrow range of offsets, should the 1D spectrum show the corresponding region as highlighted or not?

In addition to this, the slit information is not being parsed by Spectrum1D so I will need to investigate the best way to add this back in.

**Comment by pllim** [comment]
Whoops... I think I might have caused the conflict. Can you please rebase? Thanks!

**Comment by rosteen** [comment]
@astrofrog I think you missed one subset bug:

- When using the x-range tool in the 1D viewer, if I then click and drag to create a region and then drag the gray region around before the selection is finalized, it creates a bunch of new subsets instead of applying the moved selection to the subset I started out creating.

**Comment by astrofrog** [comment]
@rosteen sounds good, I'll investigate!

**Review by duytnguyendtn — COMMENTED** [review]
In reviewing these changes by @astrofrog, it works flawlessly on our internal NIRspec data! Axes look good and a Spectrum1D is successfully created! 

However, in investigating https://github.com/spacetelescope/jdaviz/pull/733#discussion_r703787773, I rediscovered the 2-axis WCS issue Tom was talking about. In digging further, the NIRspec data works because Spectrum1D is able to auto-recognize the dataset and construct the proper WCS we need. However, the NIRISS data, when I try to open it with Spectrum1D.read(), doesn't actually read it in, hence why we need to construct it manually. A comparable NIRISS 2D Spectrum reader for Spectrum1D seems to be necessary. The nice thing is I believe this will also be the missing piece to finally kill off the separate NIRISS parser, and finally reunite the two.

It seems we have a decision to make (And I'll even make plans, since it seems our team really likes them 😉 ): 

- Plan A: Go forward with this PR as it stands and be happy with at least getting the generic, non-hardcoded process, working with NIRspec. Then, after the release, work with the appropriate team (NOTE: I don't think we wrote the NIRSpec one) to write up a NIRISS Spectrum1D reader as part of the broader loader unification/generalization goal. 
- Plan B: Stall this and write the Spectrum1D reader first, then revisit this later. I will note, this will effectively mean we miss our JWST Build goals.

Thoughts? Particularly pinging @camipacifici because of the nature NIRISS. Also @PatrickOgle @orifox @larrybradley

File: jdaviz/configs/mosviz/plugins/parsers.py (hunk 1 above)
Comment (astrofrog) [reply jdaviz/configs/mosviz/plugins/parsers.py:560]: Yes we could just not add a WCS and link by pixels in this case.

**Comment by astrofrog** [comment]
Just a quick update from my side: I've released glue-astronomy v0.3.1 which fixes another of the subset issues I was seeing (selections in 1D spectrum now propagate to the 2D spectrum viewer). However, the rebase is proving tricky so I am going to have to spend a little time making sure it is done carefully - meaning I likely won't get to it until next Tuesday.

**Comment by astrofrog** [comment]
I managed to do the rebase - so the two last items to investigate are the missing slit information in the 2D spectra and the issue mentioned by @rosteen in https://github.com/spacetelescope/jdaviz/pull/733#issuecomment-915404725 (although I cannot reproduce the latter for now).

Otherwise I agree with Plan A from @duytnguyendtn in https://github.com/spacetelescope/jdaviz/pull/733#pullrequestreview-750700549 - I can't imagine that writing the parser for NIRISS data should be too difficult?

**Comment by larrybradley** [comment]
I agree with Plan A.

File: jdaviz/configs/mosviz/plugins/parsers.py (hunk 1 above)
Comment (duytnguyendtn) [reply jdaviz/configs/mosviz/plugins/parsers.py:557]: It sounds in offband conversations, the Product Owners are favoring just pushing this through. It seems Pixel Linking will only stall this PR further, so we can tackle this improvement at a later date

**Comment by rosteen** [comment]
@astrofrog I just updated my glue stack and tested with this PR and can't reproduce the error I reported, it appears to be fixed.

**Comment by astrofrog** [comment]
Great! So it's really just the region overlay now that needs to be restored I think.

**Comment by duytnguyendtn** [comment]
Thanks to Ori and Brian for doing some great investigation work for us! Here's the bombshell:

> The Cal files don’t have wavelength info.  The resampled s2d files, do. But it looks like we don’t produce those.  That means we’ll need to, indeed, think about how we want to handle 2d spectra for NIRISS and NIRCam….if at all.

After this PR, we'll need to make a decision on this one. Pixel linking? Somehow else?

**Comment by astrofrog** [comment]
Pixel linking makes sense and is the fastest if they are on the same pixel grid as the 1D spectra.

**Comment by duytnguyendtn** [comment]
Update 2: While I still think we should pursue Pixel-Linking as a fallback in ALL viz tools so we can support lots of general data, we did have another breakthrough: apparently there's a WAVELENGTH extension? We're trying to figure out how to turn this into a useful WCS axis, but it's at least a lead

temp['WAVELENGTH'].header
XTENSION= 'IMAGE   '           / Image extension                                
BITPIX  =                  -32 / array data type                                
NAXIS   =                    2 / number of array dimensions                     
NAXIS1  =                  107                                                  
NAXIS2  =                   11                                                  
PCOUNT  =                    0 / number of parameters                           
GCOUNT  =                    1 / number of groups                               
EXTNAME = 'WAVELENGTH'         / extension name                                 
EXTVER  =                    1 / extension value

**Comment by astrofrog** [comment]
Does that HDU just give the wavelength for each pixel? This would then allow the wavelength axis to be different for each row of the 2D spectrum - if so this can't strictly be represented as a Spectrum1D but we could potentially find the average wavelength for each column in the data and treat it as a 1D Tabular WCS to a first approximation.

If we wanted to treat it exactly we would need to creat some kind of 2D interpolating WCS and store the data in an NDCube instance rather than Spectrum1D and then create a translator for that.

**Comment by astrofrog** [comment]
I believe this is now ready for review and final testing!

The slit overlay works again for NIRCAM data, and I have implemented a fix for the loading of the NIRISS 2D spectrum wavelength information. I have not implemented any kind of fallback to pixel linking for now as this is beyond the scope of this PR. With this PR as-is, functionality should be the same as before but with many of the subset issues hopefully fixed and with more sensible axes for the 2D spectra:

Screenshot from 2021-09-14 14-59-52

An interesting effect that you might see is the following:

Peek 2021-09-14 14-48

This is 'correct' because the 2D spectra are actually in reverse wavelength order. This can be fixed in the 2D spectrum viewer if desired, but also beyond the scope of this PR.

**Comment by pllim** [comment]
Does this mean we can remove spectral_cube from the dependency list too?

**Comment by rosteen** [comment]
> Does this mean we can remove spectral_cube from the dependency list too?

Not until we also merge @javerbukh's PR removing it from Cubeviz.

**Comment by pllim** [comment]
Which PR is that?

**Comment by javerbukh** [comment]
@pllim #547

**Review by rosteen — APPROVED** [review]
Looks good to me, thanks! We'll definitely have to double check that everything works with #852 after whichever one gets merged first is in.

**Comment by pllim** [comment]
Who has better rebase skills: @astrofrog or @mariobuikhuizen ? 😆

**Review by javerbukh — APPROVED** [review]
Looks good and works as expected!

**Comment by astrofrog** [comment]
Merge in numerical order? 😂

**Pull request merged**